### DISCUSSION TITLE
[Feature] Adding in Sun386i support

### DISCUSSION THREAD

**Discussion by MagnetarRocket** [86Box/86Box#4277] [Ideas]
**Is this related to a existing problem?**
Not a problem, but another feature request is similar to this (did add in a comment to a comment myself): [ Request to add support for NEC PC-98 standard #3857 ](https://github.com/86Box/86Box/discussions/3857)

**Describe the solution you'd like**
Being able to boot with the Sun 386i non-BIOS system as the emulated target, while the IBM PS/2 line is support already in 86box, other non-BIOS firmware from the heydays of the 80386 & Pentium 2 are excluded. The very short list of such systems that i could find were the Sun386i & SGI Visual workstation(not likely to have the firmwares of the SGI reversed-engineered/dumped as the OS'es shipped were & still are somewhat common).

**Describe alternatives you've considered**
Truth is i don't think there are alternatives that use the firmware of said systems. The Sun386i is it's own thing i think, also it seems the Sun386i has been lost to the bowels of history. (Most sun emulation applications do m68k & SPARC)

**Additional context** 
[Winworld](https://winworldpc.com/product/sunos/4x) "SunOS 4.0.1 x86"
[http://www.bitsavers.org/pdf/sun/](http://www.bitsavers.org/pdf/sun/) _look for 386i_
_Currently there seems to be no dump of the Sun386i firmware to work off from, or it's hardware graphics frambuffers_
[NME](https://github.com/phabrics/nme), ["emulator sun/2"](https://github.com/lisper/emulator-sun-2), and [QEMU](https://github.com/qemu/qemu) to get the hang of sun firmware philosophy, i think.
Screenshot of the unique firmware: ![Sun 386i boot firmware screenshot](http://qolc.net/tihkal/acme/acme-bootscreen.jpg)

**Comment by binarymaster** [comment]
I'd be interested in SGI Visual Workstation 320/540 emulation, since this would allow testing NT boot from ARCS firmware.

**Reply by MagnetarRocket** [reply]
Was thinking about that when i was writing this feature request, but decided to do the Sun386i as it's a even rarer bird then the Visual Workstations (The very special version of SunOS for these things had a different version of sunview than every other version of SunOS, both before & after, even on the SPARC & m68k Suns).

**Reply by Cacodemon345** [reply]
The inner workings of SGI Visual Workstation 320/540 is completely undocumented, and safe to say, has realistically little to no chances to get emulated, not to mention the also completely undocumented 3D video adapter it has.

I don't think Linux even actually supports it.

**Reply by binarymaster** [reply]
According to the docs, Linux supported it, but without video acceleration:
- https://sgi320.wordpress.com/community/tutorials/installing-linux/
- https://linux-visws.sourceforge.net/
- https://web.archive.org/web/20160319081221/http://oss.sgi.com/www.linux.sgi.com/intel/

Support was dropped in https://github.com/torvalds/linux/commit/c5f9ee3d665a7660b296aa1e91949ae3376f0d07.

**Comment by thunderbird32** [comment]
Firmware for the 386i could **possibly** be sourced from [jpkiwigeek](https://www.youtube.com/@jpkiwigeek) aka [HapNz](https://www.reddit.com/user/hapnz/). I know [at one point he had a 386i](https://youtu.be/rrb9--B-AEk?si=5pqFYNlDdVZzMQ8F), although I don't know if it's still in his collection. Unfortunately he seems to have fallen off the face of the Earth.

EDIT: I took the initiative to message HapNz on Reddit just to see if he's still around. IIRC, his 386i wasn't working, but that's also been 13 years ago (according to YouTube datestamps). Maybe he got it working since then?
